cmd/update.go

```diff
@@ -1,101 +1,99 @@
+			os.Exit(1)
 		}
 
 		// get status if flag is set and exit
-		if status {
-			err := magellan.GetUpdateStatus(q)
-			if err != nil {
-				l.Log.Errorf("failed toget update status: %v", err)
+		for _, arg := range args {
+			if showStatus {
+				err := magellan.GetUpdateStatus(&magellan.UpdateParams{
+					FirmwarePath:     firmwareUrl,
+					FirmwareVersion:  firmwareVersion,
+					Component:        component,
+					TransferProtocol: transferProtocol,
+					CollectParams: magellan.CollectParams{
+						URI:      arg,
+						Username: username,
+						Password: password,
+						Timeout:  timeout,
+					},
+				})
+				if err != nil {
+					log.Error().Err(err).Msgf("failed to get update status")
+				}
+				return
 			}
-			return
-		}
 
-		// client, err := magellan.NewClient(l, &q.QueryParams)
-		// if err != nil {
-		// 	l.Log.Errorf("failed tomake client: %v", err)
-		// }
-		// err = magellan.UpdateFirmware(client, l, q)
-		err := magellan.UpdateFirmwareRemote(q)
-		if err != nil {
-			l.Log.Errorf("failed toupdate firmware: %v", err)
+			// initiate a remote update
+			err := magellan.UpdateFirmwareRemote(&magellan.UpdateParams{
+				FirmwarePath:     firmwareUrl,
+				FirmwareVersion:  firmwareVersion,
+				Component:        component,
+				TransferProtocol: strings.ToUpper(transferProtocol),
+				CollectParams: magellan.CollectParams{
+					URI:      host,
+					Username: username,
+					Password: password,
+					Timeout:  timeout,
+				},
+			})
+			if err != nil {
+				log.Error().Err(err).Msgf("failed to update firmware")
+			}
 		}
 	},
 }
 
 func init() {
-	updateCmd.Flags().StringVar(&host, "bmc-host", "", "set the BMC host")
-	updateCmd.Flags().IntVar(&port, "bmc-port", 443, "set the BMC port")
-	updateCmd.Flags().StringVar(&username, "user", "", "set the BMC user")
-	updateCmd.Flags().StringVar(&password, "pass", "", "set the BMC password")
-	updateCmd.Flags().StringVar(&transferProtocol, "transfer-protocol", "HTTP", "set the transfer protocol")
-	updateCmd.Flags().StringVar(&protocol, "protocol", "https", "set the Redfish protocol")
-	updateCmd.Flags().StringVar(&firmwareUrl, "firmware-url", "", "set the path to the firmware")
-	updateCmd.Flags().StringVar(&firmwareVersion, "firmware-version", "", "set the version of firmware to be installed")
-	updateCmd.Flags().StringVar(&component, "component", "", "set the component to upgrade")
-	updateCmd.Flags().BoolVar(&status, "status", false, "get the status of the update")
+	updateCmd.Flags().StringVar(&username, "username", "", "Set the BMC user")
+	updateCmd.Flags().StringVar(&password, "password", "", "Set the BMC password")
+	updateCmd.Flags().StringVar(&transferProtocol, "transfer-protocol", "HTTP", "Set the transfer protocol")
+	updateCmd.Flags().StringVar(&firmwareUrl, "firmware.url", "", "Set the path to the firmware")
+	updateCmd.Flags().StringVar(&firmwareVersion, "firmware.version", "", "Set the version of firmware to be installed")
```

Comment:
I think it would be better to switch these back to dashes, i.e. `firmware-url` and `firmware-version`. Mixing periods with dashes (e.g. we already have `transfer-protocol` above) could get confusing.

Reply:
I see I haven't done it in this repo, but in other repos I have flags that are meant to be represented as a object reflecting the config structure using the dot notation, such as `jwks.url` and `jwks.fetch-retries` vs `jwks-url` and `jwks-fetch-retries`. I think that this notation is less confusing when there are flags like `client.access-token` and if you're thinking about the same hierarchy as in the config. So something like `transfer.protocol` wouldn't make sense unless there is a `transfer` object being represented in the config followed by a nested `protocol` property. I had the `bmc.host` and `bmc.port` flags using this notation before removing the flags altogether.

Also a side note: I think I'm going to change `transfer-protocol` to `scheme` so it's similar to the other subcommands.

Reply:
That's fair, and I can see the reasoning for organizing options that way.

I just know that when I see the list of flags in the help menu and see `.` mixed in with `-`, I now have to remember where to put the `.` and where to put the `-` versus remembering the config words and separating them by only `-`. Like in the help output for `update`:

```
Usage:
  magellan update hosts... [flags]

Flags:
      --component string           Set the component to upgrade (BMC|BIOS)
      --firmware.url string        Set the path to the firmware
      --firmware.version string    Set the version of firmware to be installed
  -h, --help                       help for update
      --password string            Set the BMC password
      --status                     Get the status of the update
      --transfer-protocol string   Set the transfer protocol (default "HTTP")
      --username string            Set the BMC user
```

I see the `--firmware...` flags separated by `.` and it looks a bit out of place when my instinct would be to want to use `--firmware-url` and `--firmware-version`.

Also, I think it would be different if you had started with `--firmware.url` and `--firmware.version` in the first place, but you're changing it to that from the `-` version.

But perhaps that is a personal preference of mine based on what I see being convention in other projects. @alexlovelltroy, @njones-lanl, and @travisbcotton, what do you think?

Reply:
Commit [5b1e3e7](https://github.com/OpenCHAMI/magellan/pull/46/commits/5b1e3e732eb3c230d8a1ea6f98e699b375b9e922) changes the `.` back to `-` and we've decided to resolve this for now to keep things moving. If the three of you (or anyone else) mentioned above has anything to offer and we decide to change this, we can discuss it further.

For now, I'm resolving this discussion.